packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-secret-manager/auth-secret-manager.ts

```diff
@@ -0,0 +1,97 @@
+import { $TSContext, stateManager } from 'amplify-cli-core';
+import aws from 'aws-sdk';
+import { getFullyQualifiedSecretName, oAuthObjSecretKey } from './secret-name';
+
+/**
+ * Manages the state of OAuth secrets in AWS ParameterStore
+ */
+export class OAuthSecretsStateManager {
+  private static instance: OAuthSecretsStateManager;
+
+  static getInstance = async (context: $TSContext): Promise<OAuthSecretsStateManager> => {
+    if (!OAuthSecretsStateManager.instance) {
+      OAuthSecretsStateManager.instance = new OAuthSecretsStateManager(await getSSMClient(context));
+    }
+    return OAuthSecretsStateManager.instance;
+  };
+
+  private constructor(private readonly ssmClient: aws.SSM) {
+  }
+
+  /**
+ * Set OAuth secret in parameter
+ */
+  setOAuthSecrets = async (hostedUISecretObj: string, resourceName: string): Promise<void> => {
+    const { envName } = stateManager.getLocalEnvInfo();
+    const secretName = getFullyQualifiedSecretName(oAuthObjSecretKey, resourceName, envName);
+    const secretValue = hostedUISecretObj;
+    await this.ssmClient
+      .putParameter({
+        Name: secretName,
+        Value: secretValue,
+        Type: 'SecureString',
+        Overwrite: true,
+      }).promise();
+  }
+
+  /**
+   * checks if the specified OAuth secrets exists in parameter store
+   */
+  hasOAuthSecrets = async (resourceName: string): Promise<boolean> => {
+    const { envName } = stateManager.getLocalEnvInfo();
+    const secretName = getFullyQualifiedSecretName(oAuthObjSecretKey, resourceName, envName);
+    try {
+      await this.ssmClient
+        .getParameter({
+          Name: secretName,
+          WithDecryption: true,
+        })
+        .promise();
+    } catch (err) {
```

Comment:
to be completely accurate we should check that the error was a "not exists" exception rather than something else